### PR TITLE
DMP-1043/DMP-1049: Implement create and modify user account

### PR DESCRIPTION
**Pull request by chrisbellingham-hmcts**
# DMP-1043 & DMP-1049

Implement POST (create) and PATCH (modify) user operations.

Note the implementation does not apply any auth restrictions to these endpoints, this will be handled in a future task.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/darts/usermanagement/UserManagementFunctionalTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                    "email_address": "james.smith@hmcts.net",
+                    "description": "Functional test user",
+                    "state": "ENABLED",
+                    "security_groups": [ ]
+                }
+                """,
+            response.asString(),
+            new CustomComparator(
+                JSONCompareMode.NON_EXTENSIBLE,
+                new Customization("id", new RegularExpressionValueMatcher<>("\\d*"))
```
Comment (richardbull) [inline src/functionalTest/java/uk/gov/hmcts/darts/usermanagement/UserManagementFunctionalTest.java:42]: I guess this regex should really be `\\d+` 
+ matches the previous token between one and unlimited times, as many times as possible

Comment (chrisbellingham-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/darts/usermanagement/UserManagementFunctionalTest.java]: Well spotted, updated.

File: src/main/java/uk/gov/hmcts/darts/usermanagement/controller/UserController.java [hunk 2]
```diff
@@ -14,13 +20,32 @@
...
+
     @Override
     @SecurityRequirement(name = SECURITY_SCHEMES_BEARER_AUTH)
     @Authorisation(contextId = ANY_ENTITY_ID, globalAccessSecurityRoles = ADMIN)
     public ResponseEntity<List<UserWithIdAndLastLogin>> usersGet(Integer courthouse) {
         return UserApi.super.usersGet(courthouse);
     }
 
+    @Override
+    public ResponseEntity<UserWithId> usersPost(User user) {
```
[the remark this answers is not in the document]
Comment (chrisbellingham-hmcts) [reply src/main/java/uk/gov/hmcts/darts/usermanagement/controller/UserController.java:37]: Authorisation will be added on a separate ticket, as there is functional test impact. Will add `@SecurityRequirement`

File: src/main/resources/openapi/usermanagement.yaml [hunk 3]
```diff
@@ -196,6 +196,7 @@ components:
     UserName:
       minLength: 1
       maxLength: 256
+      pattern: '^\S.+$' # Disallow starting or ending with whitespace
```
[the remark this answers is not in the document]
Comment (chrisbellingham-hmcts) [reply src/main/resources/openapi/usermanagement.yaml]: Done

**Review by richardbull — APPROVED** [review]

**Pull request merged**